app/views/oaisys/pmh/list_identifiers.xml.builder

```diff
@@ -0,0 +1,33 @@
+xml.push_deferred_attribute('xmlns:dc': 'http://purl.org/dc/elements/1.1/',
+                            'xmlns:oai-id': 'http://www.openarchives.org/OAI/2.0/oai-identifier',
+                            'xmlns': 'http://www.openarchives.org/OAI/2.0/',
+                            'xmlns:etd_ms': 'http://www.ndltd.org/standards/metadata/etdms/1.0/',
+                            'xmlns:atom': 'http://www.w3.org/2005/Atom',
+                            'xmlns:rdfs': 'http://www.w3.org/2000/01/rdf-schema#',
+                            'xmlns:rdf': 'http://www.w3.org/1999/02/22-rdf-syntax-ns#',
+                            'xmlns:dcterms': 'http://purl.org/dc/terms/',
+                            'xmlns:oreatom': 'http://www.openarchives.org/ore/atom/',
+                            'xmlns:oai_dc': 'http://www.openarchives.org/OAI/2.0/oai_dc/',
+                            'xmlns:xsi': 'http://www.w3.org/2001/XMLSchema-instance',
+                            'xsi:schemaLocation': 'http://www.openarchives.org/OAI/2.0/ '\
+                            'http://www.openarchives.org/OAI/2.0/OAI-PMH.xsd '\
+                            'http://www.openarchives.org/OAI/2.0/oai-identifier '\
+                            'http://www.openarchives.org/OAI/2.0/oai-identifier.xsd '\
+                            'http://www.openarchives.org/OAI/2.0/oai_dc/ '\
+                            'http://www.openarchives.org/OAI/2.0/oai_dc.xsd '\
+                            'http://www.ndltd.org/standards/metadata/etdms/1.0/ '\
+                            'http://www.ndltd.org/standards/metadata/etdms/1-0/etdms.xsd '\
+                            'http://www.w3.org/2005/Atom')
+
+xml.tag!('request', parameters, 'https://era.library.ualberta.ca/oai')
+xml.ListIdentifiers do
+  results.each do |result|
+    xml.tag!('header') do
+      xml.identifier 'oai:era.library.ualberta.ca:' + result.id
```

Comment:
Ok, so: bit of an efficiency issue here: Boil away the abstractions and what we ended up calling was:

```ruby
# either 
Item.public_items
# or
Item.public_items.belongs_to_path(params[:set].tr(':', '/'))  
```

In either case that translates into doing a `SELECT * FROM Items WHERE...`, taking the hundred or 10s of thousands (in Production) of results, instantiating each into an instance of the Item model, passing the whole giant array of instances into the view, iterating through all of those items, and then just (essentially) printing out their ID and created data and maybe paths.

As you can imaging that eats up a lot of memory (all of the SQL results + 20,000 Item instances) and CPU (instantiating and then having the GC immediately cleaning up 20,000 Items).

Happily, there's a way more efficient way to get _just_ the information we need out of the database, without all of the extraneous columns that make up an Item that we never plan to use here:

```ruby
Item.public_items.pluck(:id, :record_created_at)
```

translates into `SELECT id, record_created_at FROM Items WHERE ...` and will return results of the form:

```ruby
[[12345, 'jan 1'], [67890, 'jan 1'], ...]
```

which is a lot lighter weight.

So we should probably be adding the `pluck` in `list_identifiers` and then alter the view to work only with the resulting array of the exact results it needs